### PR TITLE
sshportal: init at 1.18.5

### PR DESCRIPTION
**Pull request by zaninime**
###### Motivation for this change
Add package `sshportal`, a simple, fun and transparent SSH (and telnet) bastion server.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dasj19 — COMMENTED** [review]
I got the resulting patch and built it on my system...

I can see it is downloading a bunch of go packages. Doesn't this make it unreproducible if one of the go package changes version over time?

I'm not familiar with go packaging but I think there is a way to address this.
edit: go2nix looks like it was made for this purpose

**Comment by zaninime** [comment]
It uses the `buildGoModule` infra with `vendorSha256`, so dependencies are pinned. Hydra would otherwise fail.

**Review by dasj19 — APPROVED** [review]
Ok thanks for explaining. I built the package and tested the binary. It seems to work.

Also the changes look good.

File: pkgs/servers/sshportal/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "1.18.5";
+
+  src = fetchFromGitHub {
+    owner = "moul";
+    repo = "sshportal";
+    rev = "v${version}";
+    sha256 = "1d9zws7b6ng23vyk8di5jmnbsxz7kaj4z8wz43wkwr6b65c2h0bf";
+  };
+
+  ldflags = [ "-X main.GitTag=${version}" "-X main.GitSha=${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/servers/sshportal/default.nix:14]:
```suggestion
  ldflags = [ "-X main.GitTag=${version}" "-X main.GitSha=${version}" "-s" "-w" ];
```

File: pkgs/servers/sshportal/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  vendorSha256 = "0fnn455adw4bhz68dpqaii08wv7lifdd5kx95rpqxsi2sqrzr4br";
+
+  meta = with lib; {
+    description = "Simple, fun and transparent SSH (and telnet) bastion server";
+    homepage = "https://manfred.life/sshportal";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ zaninime ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/servers/sshportal/default.nix:23]:
```suggestion
```
default to linux+darwin and only works there

**Comment by zaninime** [comment]
@SuperSandro2000 suggestions applied. PTAL, thanks.

**Pull request merged**